mllib/src/main/scala/org/apache/spark/mllib/optimization/GradientDescent.scala

```diff
@@ -122,8 +122,8 @@ class GradientDescent private[spark] (private var gradient: Gradient, private va
    * @return solution vector
    */
   @DeveloperApi
-  def optimize(data: RDD[(Double, Vector)], initialWeights: Vector): Vector = {
-    val (weights, _) = GradientDescent.runMiniBatchSGD(
+  def optimize(data: RDD[(Double, Vector)], initialWeights: Vector): (Vector, Double, Integer) = {
```

Comment:
This API should not be changed.  You could add a new method (```optimizeWithStats```?) which returns the 3 values, and then share the implementation.